tests/mtd_raw/main.c

```diff
@@ -0,0 +1,458 @@
+    addr = atoi(argv[2]);
+    len  = atoi(argv[3]);
+
+    void *buffer = malloc(len);
+    if (buffer == NULL) {
+        puts("out of memory");
+        return -1;
+    }
+
+    int res = mtd_read(dev, buffer, addr, len);
+
+    od_hex_dump_ext(buffer, len, 0, addr);
+
+    free(buffer);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_read_page(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t page, offset, len;
+
+    if (oob) {
+        return -1;
+    }
+
+    if (argc < 5) {
+        printf("usage: %s <dev> <page> <offset> <len>\n", argv[0]);
+        return -1;
+    }
+
+    page   = atoi(argv[2]);
+    offset = atoi(argv[3]);
+    len    = atoi(argv[4]);
+
+    void *buffer = malloc(len);
+    if (buffer == NULL) {
+        puts("out of memory");
+        return -1;
+    }
+
+    int res = mtd_read_page(dev, buffer, page, offset, len);
+
+    od_hex_dump_ext(buffer, len, 0, page * dev->page_size + offset);
+
+    free(buffer);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_write(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t addr, len;
+
+    if (oob) {
+        return -1;
+    }
+
+    if (argc < 4) {
+        printf("usage: %s <dev> <addr> <data>\n", argv[0]);
+        return -1;
+    }
+
+    addr = atoi(argv[2]);
+    len  = strlen(argv[3]);
+
+    int res = mtd_write(dev, argv[3], addr, len);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_write_page(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t page, offset, len;
+
+    if (oob) {
+        return -1;
+    }
+
+    if (argc < 5) {
+        printf("usage: %s <dev> <page> <offset> <len>\n", argv[0]);
+        return -1;
+    }
+
+    page   = atoi(argv[2]);
+    offset = atoi(argv[3]);
+    len    = strlen(argv[4]);
+
+    int res = mtd_write_page(dev, argv[4], page, offset, len);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_erase(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t addr;
+    uint32_t len;
+
+    if (oob) {
+        return -1;
+    }
+
+    if (argc < 4) {
+        printf("usage: %s <dev> <addr> <len>\n", argv[0]);
+        return -1;
+    }
+
+    addr = atoi(argv[2]);
+    len  = atoi(argv[3]);
+
+    int res = mtd_erase(dev, addr, len);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_erase_sector(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t sector, count = 1;
+
+    if (oob) {
+        return -1;
+    }
+
+    if (argc < 3) {
+        printf("usage: %s <dev> <sector> [count]\n", argv[0]);
+        return -1;
+    }
```

Comment:
Just a very small improvement for UX: I'd move this check for all `cmd_*` methods above the `_get_dev()` call to make it easy for users to learn how the CLI is used.

Otherwise the CLI output looks something like this:
```
> help
help
Command              Description
---------------------------------------
info                 Print properties of the MTD device
power                Turn the MTD device on/off
read                 Read a region of memory on the MTD device
read_page            Read a region of memory on the MTD device (pagewise addressing)
write                Write a region of memory on the MTD device
write_page           Write a region of memory on the MTD device (pagewise addressing)
erase                Erase a region of memory on the MTD device
erase_sector         Erase a sector of memory on the MTD device
test                 Erase & write test data to the last two sectors
reboot               Reboot the node
version              Prints current RIOT_VERSION
pm                   interact with layered PM subsystem
> write
write
write: please specify the MTD device
> write 0
write 0
usage: write <dev> <addr> <data>
> write 0 0 hello
write 0 0 hello
> read
read
read: please specify the MTD device
> read 0
read 0
usage: read <dev> <addr> <len>
> read 0 0 5
read 0 0 5
00000000  68  61  6C  6C  6F
```